### PR TITLE
Framework: Make JSON pretty too

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

* Prettier .json on pre-commit
* Don't eslint package.json on pre-commit
* Prettier .json on reformat files

#### Testing instructions

* Modify a .json file with bad fromatting
* commit
* Watch it be reformatted

### CODE REVIEW

File: bin/pre-commit-hook.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ function parseGitDiffToPathArray( command ) {
 	return execSync( command, { encoding: 'utf8' } )
 		.split( '\n' )
 		.map( name => name.trim() )
-		.filter( name => /\.(jsx?|scss)$/.test( name ) );
+		.filter( name => /(?:package.json|\.jsx?|\.scss)$/.test( name ) );
```
Comment (simison) [inline bin/pre-commit-hook.js]: What do you think about including all *.json files here?

Comment (sirreal) [inline bin/pre-commit-hook.js]:
```suggestion
		.filter( name => /(?:\.json|\.jsx?|\.scss)$/.test( name ) );
```

File: package.json [hunk 2]
```diff
@@ -255,7 +255,7 @@
 		"prestart": "npx check-node-version --package && npm run -s install-if-deps-outdated && node bin/welcome.js",
 		"start": "npm run -s build",
 		"poststart": "cross-env-shell CALYPSO_CLIENT=true NODE_PATH=$NODE_PATH:server:client:. node $NODE_ARGS --max_old_space_size=4096 build/bundle.js",
-		"reformat-files": "./node_modules/.bin/prettier --ignore-path .eslintignore --write \"**/*.{js,jsx}\"",
+		"reformat-files": "./node_modules/.bin/prettier --ignore-path .eslintignore --write \"**/*.{js,jsx}\" \"**/package.json\"",
```
Comment (sirreal) [inline package.json]:
```suggestion
		"reformat-files": "./node_modules/.bin/prettier --ignore-path .eslintignore --write \"**/*.{js,jsx,json}\"",
```

**Comment by sirreal** [comment]
Updated to prettier all the json

**Comment by sirreal** [comment]
Rebased!

**Review by ockham — APPROVED** [review]
:heart: it! :ship: it :smile:

**Pull request merged**